### PR TITLE
refactors some of the components to be more modular and more DRY

### PR DESCRIPTION
**Pull request by dittonjs**

### CODE REVIEW

File: client/js/components/scorm/uploader.jsx [hunk 1]
```diff
@@ -4,84 +4,70 @@ import React                    from 'react';
...
+      return null;
     }
+  }
 
-    componentWillReceiveProps(nextProps){
+  renderProgress() {
+    if (this.props.error) {
+      return (
+        <div className="c-progress-bar red">
+          <span style={{width: "100%"}}></span>
```
Comment (bfcoder) [inline client/js/components/scorm/uploader.jsx]: Use a `getStyles` instead?

Comment (dittonjs) [inline client/js/components/scorm/uploader.jsx]: yeah i should do that.

File: client/js/components/common/file_upload.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+import React from 'react';
+
+export default (props) => {
```
Comment (BenHSmith13) [inline client/js/components/common/file_upload.jsx]: We should add propTypes

File: client/js/components/common/svg_button.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+import React from 'react';
+import CommonSvg from './common_svg';
+
+export default (props) => {
```
Comment (BenHSmith13) [inline client/js/components/common/svg_button.jsx]: ditto on the PropTypes

File: client/js/components/common/common_svg.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+import React from 'react';
+
+export default (props)=>{
```
Comment (BenHSmith13) [inline client/js/components/common/common_svg.jsx]: Proptypes?

File: client/js/components/common/file_upload.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      <div className="c-btn c-btn--upload">
+        <svg className="c-icon-upload" xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
+          <path d="M0 0h48v48h-48z" fill="none"/>
+          <path className="c-path" d="M38.71 20.07c-1.36-6.88-7.43-12.07-14.71-12.07-5.78 0-10.79 3.28-13.3 8.07-6.01.65-10.7 5.74-10.7 11.93 0 6.63 5.37 12 12 12h26c5.52 0 10-4.48 10-10 0-5.28-4.11-9.56-9.29-9.93zm-10.71 5.93v8h-8v-8h-6l10-10 10 10h-6z"/>
+        </svg>
+        <input
+          id="upload"
+          type="file"
+          onChange={(e) => handleChange(e)}
+          style={inputStyles}/>
```
Comment (BenHSmith13) [inline client/js/components/common/file_upload.jsx]: Always supposed to me display none?

**Comment by bfcoder** [comment]
Specs?

File: client/js/components/scorm/course.jsx [hunk 6]
```diff
@@ -2,61 +2,26 @@
...
-    this.handleGoClick = this.handleGoClick.bind(this);
+    this.state = {
+      isGoBtnActive: false,
+      selectVal: "0",
+      isGradeActive: false
+    };
   }
 
   static propTypes = {
     course: React.PropTypes.object.isRequired
```
Comment (BenHSmith13) [inline client/js/components/scorm/course.jsx]: Victory! What about the rest of the props?

File: client/js/components/scorm/course.jsx [hunk 7]
```diff
@@ -71,61 +36,36 @@ export default class Course extends React.Component {
...
-                                                              </select>
-                                                              <svg className="c-icon-drop" xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48">
-                                                                <path className="c-path" d="M14 20l10 10 10-10z"/>
-                                                                <path d="M0 0h48v48h-48z" fill="none"/>
-                                                              </svg>
-                                                            </div>
-                                                            <button className={"c-btn c-btn--go is-" + this.state.isGoBtnActive} onClick={ (e)=>{this.handleGoClick() }}>Go</button>
-                                                          </div>
-                                                        : <div className="c-list-item__type" style={{minWidth: "20rem"}}>{this.state.selectVal}</div>;
+    let dropdownSection;
```
Comment (BenHSmith13) [inline client/js/components/scorm/course.jsx]: you can one line these, or not, your call;

File: client/js/components/scorm/scorm_index.jsx [hunk 8]
```diff
@@ -1,58 +1,15 @@
...
+import React                                    from 'react';
+import { connect }                              from 'react-redux';
+import * as ScormActions                        from '../../actions/scorm';
+import CoursesList                              from './courses_list';
+import Uploader                                 from './uploader';
+import { create_assignment, delete_assignment } from '../../libs/canvas/constants/assignments';
+import canvasRequest                            from '../../libs/canvas/action';
+import FileUpload                               from '../common/file_upload';
+
+export class ScormIndex extends React.Component {
```
Comment (BenHSmith13) [inline client/js/components/scorm/scorm_index.jsx]: proop types

File: client/js/components/scorm/import_type_selector.jsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+import React     from 'react';
+import CommonSvg from '../common/common_svg';
+
+export default (props)=>{
```
Comment (BenHSmith13) [inline client/js/components/scorm/import_type_selector.jsx]: And spec for new components

File: client/js/components/common/common_svg.jsx (hunk 4 above)
Comment (BenHSmith13) [inline client/js/components/common/common_svg.jsx]: spec for new component

File: client/js/components/common/file_upload.jsx (hunk 2 above)
Comment (BenHSmith13) [inline client/js/components/common/file_upload.jsx]: spacks

File: client/js/components/scorm/uploader.jsx [hunk 10]
```diff
@@ -4,84 +4,70 @@ import React                    from 'react';
...
-    error: state.scorm.uploadError
-  };
-};
+export class Uploader extends React.Component {
 
-@connect(select, ScormActions, null, { withRefs: true })
-export default class Uploader extends React.Component {
-    constructor(props) {
-        super(props);
+  renderError() {
```
Comment (BenHSmith13) [inline client/js/components/scorm/uploader.jsx]: own file?

File: client/js/components/scorm/uploader.jsx [hunk 11]
```diff
@@ -4,84 +4,70 @@ import React                    from 'react';
...
+        </div>
+      );
+    } else {
+      return null;
     }
+  }
 
-    componentWillReceiveProps(nextProps){
+  renderProgress() {
+    if (this.props.error) {
```
Comment (BenHSmith13) [inline client/js/components/scorm/uploader.jsx]: own file?

File: client/styles/partials/_header.scss [hunk 12]
```diff
@@ -1,5 +1,5 @@
 .c-header{
-  padding: 0 0 1.5rem;
+  padding: 1.5rem 0 1.5rem 0;
```
Comment (BenHSmith13) [inline client/styles/partials/_header.scss]: also why changing styles in this PR?

**Comment by BenHSmith13** [comment]
David updated all the file paths, this PR will need some lovin' before it's ready

File: client/js/attendance/components/layout/index.jsx [hunk 13]
```diff
@@ -3,20 +3,12 @@
 import React                    from "react";
 import { connect }              from "react-redux";
```
Comment (BenHSmith13) [inline client/js/attendance/components/layout/index.jsx]: Is connect still used?

**Comment by BenHSmith13** [comment]
Looks Good

LGTM

**Pull request merged**